File: Editor/src/main/java/com/rspsi/studio/plugin/builtin/tool/MultiSelectToolPlugin.java
package com.rspsi.studio.plugin.builtin.tool;

import com.rspsi.studio.plugin.StudioToolPlugin;
import com.rspsi.studio.theme.StudioIcons;
import com.rspsi.studio.ui.StudioPanelContext;

import java.util.EnumSet;
import java.util.Set;

/**
 * Modal tool for dragging a rectangular marquee over many tiles for inspection. Lives on the
 * floating tool rail only; the Tile Inspector panel shows what it selects.
 */
public final class MultiSelectToolPlugin implements StudioToolPlugin {

    public static final String ID = "studio.tool.select.multi";
    public static final String ENGINE_TOOL_ID = "selection.multi";

    @Override
    public String id() {
        return ID;
    }

    @Override
    public String name() {
        return "Multi Select";
    }

    @Override
    public String description() {
        return "Click and drag to marquee-select a rectangular region of tiles for inspection.";
    }

    @Override
    public String icon() {
        return StudioIcons.AREA;
    }

    @Override
    public String toolId() {
        return ENGINE_TOOL_ID;
    }

    @Override
    public String shortcut() {
        return "M";
    }

    @Override
    public int railPriority() {
        return 11;
    }

    @Override
    public String category() {
        return "Selection";
    }

    @Override
    public Set<ToolSurface> surfaces() {
        // The docked Left Tool Rail is brush-only now (Tile Painter, Height
        // Sculptor); selection lives on the floating rail instead.
        return EnumSet.of(ToolSurface.FLOATING_TOOLBAR);
    }

    @Override
    public boolean hasContextDrawerContent() {
        return false;
    }

    @Override
    public void renderContextDrawer(StudioPanelContext context) {
        // No drawer content - selection results show in the Tile Inspector panel instead.
    }
}

File: Editor/src/main/kotlin/com/rspsi/studio/plugin/builtin/tool/MultiSelectToolPlugin.kt
package com.rspsi.studio.plugin.builtin.tool

import com.rspsi.studio.plugin.StudioToolPlugin
import com.rspsi.studio.theme.StudioIcons
import com.rspsi.studio.ui.StudioPanelContext
import java.util.EnumSet

/**
 * Native Studio projection for marquee-selecting a rectangular tile region.
 *
 * Like Single Select, this is a selection-mode switcher rather than a brush tool.
 * It therefore stays on the floating selection toolbar and leaves result display
 * to the Tile Inspector instead of creating duplicate drawer UI.
 */
class MultiSelectToolPlugin : StudioToolPlugin {
    override fun id(): String = ID

    override fun name(): String = "Multi Select"

    override fun description(): String =
        "Click and drag to marquee-select a rectangular region of tiles for inspection."

    override fun icon(): String = StudioIcons.AREA

    override fun toolId(): String = ENGINE_TOOL_ID

    override fun shortcut(): String = "M"

    override fun railPriority(): Int = 11

    override fun category(): String = "Selection"

    override fun surfaces(): MutableSet<StudioToolPlugin.ToolSurface> =
        EnumSet.of(StudioToolPlugin.ToolSurface.FLOATING_TOOLBAR)

    override fun hasContextDrawerContent(): Boolean = false

    override fun renderContextDrawer(context: StudioPanelContext?) {
        // Intentionally empty: the Tile Inspector owns selection-result presentation.
    }

    companion object {
        const val ID = "studio.tool.select.multi"
        const val ENGINE_TOOL_ID = "selection.multi"
    }
}
